xskillscore/core/comparative.py

```diff
@@ -167,3 +177,141 @@ def _categorical_metric(observations, forecasts, dim):
     walk.coords["alpha"] = alpha
     walk = _add_as_coord(walk, confidence, "confidence")
     return walk
+
+
+def mae_test(
+    forecasts1,
+    forecasts2,
+    observations=None,
+    dim=[],
```

Comment:
Should be `None`, then `if dim is None: dim = []`. See mutable default arguments: https://docs.python-guide.org/writing/gotchas/.

Reply:
dim=None means all dims. See xr.mean

Reply:
The point is that `dim=[]` is bad syntax for python and causes unintended consequences. So either we make `dim=None` (all dims) the default or have some other way to target `[]` without having it in the default kwargs.

Reply:
dim=None all dims won’t work. Because this include time_dim. I could exclude that.

Reply:
I guess you are afraid of mutable defaults https://docs.python-guide.org/writing/gotchas/ but here dim is not changed inside the function